File: src/ts/src/ads/internal/MediationManager.ts
import { AdResult } from "../IFullScreenAd";
import { AsyncHelper } from "./AsyncHelper";
import { IAsyncHelper } from "./IAsyncHelper";

export class MediationManager {
    private static _sharedInstance = new MediationManager();

    private _adDisplayer: IAsyncHelper<AdResult>;

    public static getInstance(): MediationManager {
        return this._sharedInstance;
    }

    private constructor() {
        this._adDisplayer = new AsyncHelper<AdResult>();
    }

    public get adDisplayer(): IAsyncHelper<AdResult> {
        return this._adDisplayer;
    }
}

File: src/cpp/ee/ads/internal/MediationManager.cpp

//
//  MediationManager.cpp
//  ee_x
//
//  Created by Zinge on 10/23/17.
//
//

#include "ee/ads/internal/MediationManager.hpp"

#include "ee/ads/internal/AsyncHelper.hpp"

namespace ee {
namespace ads {
using Self = MediationManager;

Self& Self::getInstance() {
    static Self sharedInstance;
    return sharedInstance;
}

Self::MediationManager() {
    adDisplayer_ = std::make_shared<AsyncHelper<AdResult>>();
}

Self::~MediationManager() {}

const std::shared_ptr<IAsyncHelper<AdResult>>& Self::getAdDisplayer() const {
    return adDisplayer_;
}
} // namespace ads
} // namespace ee
